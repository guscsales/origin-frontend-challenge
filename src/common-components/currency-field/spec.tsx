import * as React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import theme from '../../configs/styles/themes';
import { ThemeProvider } from 'styled-components';
import CurrencyField, { handleChangeValue } from '.';

jest.mock('react-currency-input');

describe('CurrencyField', () => {
  const getWrapper = (props: any = {}, renderType: any = mount) =>
    renderType(
      <ThemeProvider theme={theme}>
        <CurrencyField label="test" {...props} />
      </ThemeProvider>
    );

  describe('#handleChangeValue', () => {
    it('should set value on state and calls `onChange` param', () => {
      const setValue = jest.fn();
      const onChange = jest.fn();
      const finalMaskedValue = '100';
      const finalValue = 100;

      handleChangeValue(finalMaskedValue, finalValue, setValue, onChange);

      expect(setValue).toHaveBeenCalledWith(finalValue);
      expect(onChange).toHaveBeenCalledWith(finalMaskedValue, finalValue);
    });

    it('should foward zero values when value is less or equals zero', () => {
      const setValue = jest.fn();
      const onChange = jest.fn();
      const finalMaskedValue = '';
      const finalValue = 0;

      handleChangeValue(finalMaskedValue, finalValue, setValue, onChange);

      expect(setValue).toHaveBeenCalledWith(0);
      expect(onChange).toHaveBeenCalledWith('0', 0);
    });
  });

  describe('#render', () => {
    it('should match a snapshot', () => {
      const wrapper = getWrapper({}, create).toJSON();

      expect(wrapper).toMatchSnapshot();
    });

    it('should return a component mocked rendered class', () => {
      const wrapper = getWrapper();

      expect(wrapper.exists()).toBe(true);
      expect(wrapper.find('.react-currency-input')).toHaveLength(1);
    });
  });
});
