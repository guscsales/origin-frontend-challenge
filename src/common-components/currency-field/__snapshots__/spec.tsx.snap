// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyField #render should match a snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  color: #1e1f21;
  font-size: 1.6rem;
  font-weight: 500;
  margin-bottom: 0.5rem;
  display: block;
}

.c2 {
  width: 5.3rem;
  height: 5.3rem;
  border-radius: 5px 0 0 5px;
  border: 1px solid #e1e8ed;
  background-color: #f4f8fa;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 2.3rem;
  color: #657786;
}

<div>
  <label
    className="c0"
  >
    test
  </label>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      $
    </div>
    <div
      className="react-currency-input"
    />
  </div>
</div>
`;
